core/domain/question_domain.py

```diff
@@ -229,3 +236,39 @@ def update_question_data(self, question_data):
             question_data: dict. A dict representing the question data.
         """
         self.question_data = question_data
+
+    def get_skills(self):
+        """Fetches the skills associated with the question.
+
+        Returns:
+            list(CollectionSkill). A list of CollectionSkill domain objects.
+        """
+        collection = collection_services.get_collection_by_id(
+            self.collection_id)
+        skills = collection.skills
+
+        question_skills = []
+        for skill_id in skills:
+            if self.question_id in collection.skills[skill_id].question_ids:
+                question_skills.append(collection.skills[skill_id])
+        return question_skills
+
+    def can_user_answer_question(self, user_id, collection_id):
+        """Verifies if the given user has the acquired
+        skills to answer the question.
+
+        Args:
+            user_id: str. The id of the user.
+            collection_id: str. The id of the collection.
+
+        Returns:
+            bool. A boolean representing True or False.
+        """
+        question_skills = self.get_skills()
+        user_skills = (
```

Comment:
Here and elsewhere, be very careful about names. I get the feeling we're confusing "skill_id" and "skill" all over the place. We should use skill_id when we mean skill_id.

Please check other places thoroughly as well. Otherwise we're going to have a lot of confusion.

Reply:
Updated.